opencog/atoms/core/Variables.cc

```diff
@@ -238,43 +250,41 @@ bool VarScraper::less_than(const Handle& lh, const Handle& rh,
 		// Both atoms have same arity, sort by outgoings
 		const HandleSeq& lout = lh->getOutgoingSet();
 		const HandleSeq& rout = rh->getOutgoingSet();
-		return is_ordered_link(lh)? less_than_ordered(lout, rout, ctx)
-			: less_than_unordered(lout, rout, ctx);
+		return is_ordered_link(lh) ? less_than_ordered_outgoing(lout, rout)
+			: less_than_unordered_outgoing(lout, rout);
 	}
 
-	// Both atoms are constant nodes of the same type, sort by regular
-	// atom order.
+	// None are variables, sort by node content.
 	if (not is_variable(lh))
 		return lh < rh;
 
-	// Both atoms are variables. If they are have different counts,
-	// sort by counts, as that affects their semantics.
-	unsigned lc = _fvc.get(lh);
-	unsigned rc = _fvc.get(rh);
-	if (lc != rc)
-		return lc < rc;
+	const auto& lps = _paths.at(lh);
+	const auto& rps = _paths.at(rh);
+	if (lps != rps)
+		return lps < rps;
 
-	// Both atoms are variables with same count, sort by regular atom
-	// order.
+	// Still a tie? These 2 variables are semantically equivalent. Sort
+	// by atom order then.
 	return lh < rh;
 }
 
-bool VarScraper::less_than_ordered(const HandleSeq& lhs, const HandleSeq& rhs,
-                                   const Context& ctx) const
+bool VarScraper::less_than_ordered_outgoing(const HandleSeq& lhs,
+                                            const HandleSeq& rhs) const
 {
 	OC_ASSERT(lhs.size() == rhs.size());
 	for (std::size_t i = 0; i < lhs.size(); i++)
 		if (not content_eq(lhs[i], rhs[i]))
-			return less_than(lhs[i], rhs[i], ctx);
+			return less_than(lhs[i], rhs[i]);
 	return false;
 }
 
-bool VarScraper::less_than_unordered(const HandleSeq& lhs, const HandleSeq& rhs,
-                                     const Context& ctx) const
+bool VarScraper::less_than_unordered_outgoing(const HandleSeq& lhs,
+                                              const HandleSeq& rhs) const
 {
-	return less_than_ordered(sorted(lhs, ctx), sorted(rhs, ctx), ctx);
+	return less_than_ordered_outgoing(sorted(lhs), sorted(rhs));
 }
 
+// TODO: maybe support context
```

Comment:
How is Context different from `PatternTerm`? So, `PatternTerm` already has quoting support...(added by you)

Reply:
As it turns out context, quotations, etc, are not needed here, the only thing that matters is the semantic differences between variables, and this is entirely captured by `Path`.